packages/console/src/pages/ApplicationDetails/SamlApplicationDetailsContent/AttributeMapping.tsx

```diff
@@ -0,0 +1,216 @@
+      isDirty={isDirty}
+      onSubmit={onSubmit}
+      onDiscard={reset}
+    >
+      <FormCard
+        title="application_details.saml_app_attribute_mapping.title"
+        description="application_details.saml_app_attribute_mapping.description"
+        learnMoreLink={{
+          // TODO: update this link once docs is ready
+          href: getDocumentationUrl('/connectors/enterprise-connectors'),
+          targetBlank: 'noopener',
+        }}
+      >
+        <table className={styles.table}>
+          <thead className={styles.header}>
+            <tr className={styles.row}>
+              <th>
+                <DynamicT forKey="application_details.saml_app_attribute_mapping.col_logto_claims" />
+              </th>
+              <th>
+                <DynamicT forKey="application_details.saml_app_attribute_mapping.col_sp_claims" />
+              </th>
+              <th />
+            </tr>
+          </thead>
+          <tbody className={styles.body}>
+            {formValues.map(([key, _], index) => {
+              return (
+                // eslint-disable-next-line react/no-array-index-key
+                <tr key={index} className={styles.row}>
+                  <td>
+                    {key === 'id' ? (
+                      <CopyToClipboard displayType="block" variant="border" value={key} />
+                    ) : (
+                      <Controller
+                        control={control}
+                        name={`${keyPrefix}.${index}.0` as const}
+                        render={({ field: { onChange, value } }) => (
+                          <Select
+                            isSearchEnabled
+                            value={value}
+                            options={[
+                              ...availableKeys.map((claim) => ({
+                                title: camelCaseToSentenceCase(claim),
+                                value: claim,
+                              })),
+                              // If this is not specified, the component will fail to render the current value. The current value has been excluded in `availableKeys`.
+                              { value, title: camelCaseToSentenceCase(value) },
+                            ]}
+                            onChange={(value) => {
+                              onChange(value);
+                            }}
+                          />
+                        )}
+                      />
+                    )}
+                  </td>
+                  <td>
+                    <TextInput {...register(`${keyPrefix}.${index}.1`)} />
+                  </td>
+                  <td>
+                    {key !== 'id' && (
+                      <IconButton
+                        onClick={() => {
+                          const currentValues = [...formValues];
+                          // eslint-disable-next-line @silverhand/fp/no-mutating-methods
+                          currentValues.splice(index, 1);
```

Comment:
Use `slice` instead of the mutating `splice`

Reply:
Updated.